app/src/main/java/be/ppareit/swiftp/FsService.java

```diff
@@ -111,7 +113,12 @@ public static void start() {
         Context context = App.getAppContext();
         Intent serverService = new Intent(context, FsService.class);
         if (!FsService.isRunning()) {
-            context.startService(serverService);
+            //https://developer.android.com/about/versions/oreo/background
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                context.startForegroundService(serverService);
+            } else {
+                context.startService(serverService);
```

Comment:
Can be replaced with [ContextCompat.startForegroundService()](https://developer.android.com/reference/android/support/v4/content/ContextCompat.html#startForegroundService(android.content.Context,%20android.content.Intent))